File: DSA/Shruti-Pandey/Day_2_q1.java
public class ClassCancellation {
    public static String shouldCancelClass(int threshold, int[] arrivalTimes) {
        int onTimeStudents = 0;
        
        for (int time : arrivalTimes) {
            if (time <= 0) {
                onTimeStudents++;
            }
        }
        
        return (onTimeStudents < threshold) ? "YES" : "NO";
    }

    public static void main(String[] args) {
        int threshold = 3;
        int[] arrivalTimes = {-1, -3, 4, 2};
        
        String result = shouldCancelClass(threshold, arrivalTimes);
        System.out.println(result);
    }
}

File: DSA/Hima-Swarupa-Padala/day2q1.py
def check_class_cancellation(threshold, arrival_times):
    on_time_students = [time for time in arrival_times if time <= 0]
    return "YES" if len(on_time_students) < threshold else "NO"

threshold = 4
arrival_times = [-2, -1, 0, 1, 2, 3]
print(check_class_cancellation(threshold, arrival_times))  
